src/app/dataset-view/dataset.service.ts

```diff
@@ -49,19 +53,22 @@ export class AppDatasetService {
     }
 
     public requestDatasetMainData(info: DatasetInfo): void {
-        this.datasetApi
-            .getDatasetMainData(info)
-            .subscribe((data: GetDatasetMainDataQuery) => {
-                if (isNil(data.datasets.byOwnerAndName)) {
-                    throw new Error("Dataset not resolved by ID");
+        this.datasetApi.getDatasetMainData(info).subscribe(
+            (data: GetDatasetMainDataQuery) => {
+                if (data.datasets.byOwnerAndName) {
+                    this.datasetUpdate(data.datasets.byOwnerAndName);
+                    this.overviewTabDataUpdate(data);
+                    this.dataTabDataUpdate(data);
+                    this.metadataTabDataUpdate(data);
+                    this.lineageTabDataUpdate(data);
+                } else {
+                    this.errorService.processError(new DatasetNotFoundError());
```

Comment:
I'm OK to try out this visitor / double-dispatch pattern, but my only concern that as this PR stands this pattern is not utilized.

Every error is reported as:
```
this.errorService.processError(new DatasetNotFoundError(...));
```
so we might've as well called directly:
```
this.errorSergive.onDatasetNotFound(...);
```
Visitor will only be useful when error propagates up the chain and type information is lost at the place where it's being processed. Can you think of a place where this would happen?

Reply:
I don't agree. The visitor seems very useful in a scenario where we would want to have multiple handling strategies attached, possibly overlapping by exception type. Anyway, we've started overengineering the error handling more than a week ago. Even the step of introducing error hierarchy was not well motivated, as we haven't solved any practical problem.

Reply:
@sergiimk The need in the visitor immediately paid off after introducing the custom global error handler. where one no longer has knowledge on the error type.